activerecord/lib/active_record/relation/where_clause.rb

```diff
@@ -25,9 +25,27 @@ def merge(other)
       end
 
       def except(*columns)
+        offset = 0
+
         WhereClause.new(
-          predicates_except(columns),
-          binds_except(columns),
+          *predicates.each_with_object([[], []]) { |node, (prts, bvs)|
```

Comment:
This method is too complicated. Can we refactor it so the code is more intention revealing?

Like what is `prts` and `bvs`?

What this loop is doing?

What that case is trying to find?

What that `unless` block is doing?